gson/src/main/java/com/google/gson/NewlineStyle.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gson;
+
+/**
+ * An enumeration that defines the kind of newline to use for serialization.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Newline">Wikipedia Newline article</a>
+ *
+ * @since $next-version$
+ */
+public enum NewlineStyle {
+  /**
+   * Using this style will result in the same kind of newline that the current environment uses.
+   *
+   * <p>So it will produce {@code "\r\n"} when running on Windows, and {@code "\n"} when running
+   * on macOS &amp; Linux.</p>
+   *
+   * @see System#lineSeparator()
+   */
+  CURRENT_OS(""),
```

Comment:
What about initializing this with `System.lineSeparator()` instead of `""`? That would avoid the special casing in `Gson` and would also lead to a more correct result for `getValue()`.

Or is there a reason why you did not do that?

Reply:
Done.

> Or is there a reason why you did not do that?

Because it didn't cross my mind :-)